pallets/msa/src/lib.rs

```diff
@@ -1321,28 +1322,41 @@ impl<T: Config> Pallet<T> {
 	///
 	pub fn get_granted_schemas_by_msa_id(
 		delegator: DelegatorId,
-		provider: ProviderId,
-	) -> Result<Option<Vec<SchemaGrant<SchemaId, BlockNumberFor<T>>>>, DispatchError> {
-		let provider_info =
-			Self::get_delegation_of(delegator, provider).ok_or(Error::<T>::DelegationNotFound)?;
+		provider: Option<ProviderId>,
+	) -> Result<Vec<DelegationResponse<SchemaId, BlockNumberFor<T>>>, DispatchError> {
+		let delegations = match provider {
+			Some(provider_id) => vec![(
+				provider_id,
+				Self::get_delegation_of(delegator, provider_id)
+					.ok_or(Error::<T>::DelegationNotFound)?,
+			)],
+			None => DelegatorAndProviderToDelegation::<T>::iter_prefix(delegator).collect(),
+		};
 
-		let schema_permissions = provider_info.schema_permissions;
-		if schema_permissions.is_empty() {
-			return Err(Error::<T>::SchemaNotGranted.into());
-		}
+		let mut result = vec![];
+		for (provider_id, provider_info) in delegations {
+			let schema_permissions = provider_info.schema_permissions;
+			if schema_permissions.is_empty() {
+				return Err(Error::<T>::SchemaNotGranted.into());
+			}
```

Comment:
Returning an error here made sense when this could only be called for a specific provider, but when called with `None`, I don't think we would want to error here...

Reply:
Sure, applied the change